src/utils/hooks/useFetchImportCandidatesQuery.ts

```diff
@@ -0,0 +1,37 @@
+import { keepPreviousData, useQuery } from '@tanstack/react-query';
+import { useTranslation } from 'react-i18next';
+import { FetchImportCandidatesParams, fetchImportCandidates } from '../../api/searchApi';
+import { ROWS_PER_PAGE_OPTIONS } from '../constants';
+import { useImportCandidatesParams } from './useImportCandidatesParams';
+
+const defaultYearParam = new Date().getFullYear();
+
+export const useFetchImportCandidatesQuery = (enabled = true) => {
+  const { t } = useTranslation();
+  const importCandidateQueryParams = useImportCandidatesParams();
+
+  const importCandidateParams = {
+    aggregation: 'all',
+    query: importCandidateQueryParams.queryParam,
+    publicationYear: importCandidateQueryParams.publicationYearParam ?? defaultYearParam,
+    files: importCandidateQueryParams.filesParam,
+    importStatus: importCandidateQueryParams.importStatusParam ?? 'NOT_IMPORTED',
+    orderBy: importCandidateQueryParams.orderByParam ?? 'createdDate',
+    sortOrder: importCandidateQueryParams.sortOrderParam ?? 'desc',
+    topLevelOrganization: importCandidateQueryParams.topLevelOrganizationParam,
+    type: importCandidateQueryParams.typeParam,
+    collaborationType: importCandidateQueryParams.collaborationTypeParam,
+    size: importCandidateQueryParams.sizeParam ?? ROWS_PER_PAGE_OPTIONS[0],
+    from: importCandidateQueryParams.fromParam ?? 0,
+  } satisfies FetchImportCandidatesParams;
```

Comment:
Om noen lurer på hva `satisifies` er:

Alternativet her er å bruke en av disse: 
```
const importCandidateParams = {...} as FetchImportCandidatesParams`;
eller 
const importCandidateParams: FetchImportCandidatesParams = {...};
```
Disse to vil gjøre det samme, som er å sikre at typen vil være nøyaktig som `FetchImportCandidatesParams`. I den typen er alle felter nullable, feks `from: number | null`, som gjør at typen her også vil si at `from` kan være `null`, selv om den defaulter til `0` på linje 25, så vi vet jo egentlig at den aldri vil være `null`.

`satisfies` er litt annerledes ved at den bare sier at verdien tilfredsstiller alle kravene til angitt type, men den kan også legge til mer info om ting vi vet om typen. Feks i dette tilfellet vil alle verdier som har en fallback ikke lenger være nullable. Da slipper vi i ha mange nullsjekker for `from` og alle de andre feltene når de skal brukes senere.

Ganske nylig jeg ble obs over `satisfies`, men jeg liker den veldig godt, og tror vi med fordel burde bruke den mye mer

Reply:
Kult!